examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -0,0 +1,728 @@
+        pos_label=pos_label,
+        linestyle=linestyle,
+        color=color,
+        ax=axs[1],
+        name=name,
+        plot_chance_level=idx == 1,
+    )
+    axs[1].plot(
+        scoring["fpr"](est, X_test, y_test),
+        scoring["tpr"](est, X_test, y_test),
+        marker,
+        markersize=10,
+        color=color,
+        label=f"Cut-off point at probability of {decision_threshold:.2f}",
+    )
+
+axs[0].set_title("Precision-Recall curve")
+axs[0].legend()
+axs[1].set_title("ROC curve")
+axs[1].legend()
+
+axs[2].plot(
+    tuned_model.cv_results_["thresholds"],
+    tuned_model.cv_results_["scores"],
+    color="tab:orange",
+)
+axs[2].plot(
+    tuned_model.best_threshold_,
+    tuned_model.best_score_,
+    "o",
+    markersize=10,
+    color="tab:orange",
+    label="Optimal cut-off point for the business metric",
+)
+axs[2].legend()
+axs[2].set_xlabel("Decision threshold (probability)")
+axs[2].set_ylabel("Objective score (using cost-matrix)")
+axs[2].set_title("Objective score as a function of the decision threshold")
+
+_ = fig.suptitle("Tuned GBDT model without refitting and using the entire dataset")
+
+# %%
+# Regarding the cut-off point, we observe that the optimum is similar to the multiple
+# repeated cross-validation case. However, be aware that a single split does not account
+# for the variability of the fit/predict process and thus we are unable to know if there
+# is any variance in the cut-off point. The repeated cross-validation averages out
+# this effect.
+#
+# Another observation concerns the ROC and Precision-Recall curves of the tuned model.
+# As expected, these curves differ from those of the vanilla model, given that we
+# trained the underlying classifier on a subset of the data provided during fitting and
+# reserved a validation set for tuning the cut-off point.
+#
+# Cost-sensitive learning when gains and costs are not constant
+# -------------------------------------------------------------
+#
+# As stated in [2]_, gains and costs are generally not constant in real-world problems.
+# In this section, we use a similar example as in [2]_ by using credit cards
+# records.
+#
+# The credit card dataset
+# ^^^^^^^^^^^^^^^^^^^^^^^
+credit_card = fetch_openml(data_id=1597, as_frame=True, parser="pandas")
+credit_card.frame.info()
+
+# %%
+# The dataset contains information about credit card records from which some are
+# fraudulent and others are legitimate. The goal is therefore to predict whether or
+# not a credit card record is fraudulent.
+#
+# In addition, we have extra information regarding the amount of each card transaction.
+# This information is used to define the business metric later.
+columns_to_drop = ["Class", "Amount"]
+data = credit_card.frame.drop(columns=columns_to_drop)
+target = credit_card.frame["Class"].astype(int)
+amount = credit_card.frame["Amount"].to_numpy()
+
+# %%
+# First, we check the class distribution of the datasets.
+target.value_counts(normalize=True)
```

Comment:
I think you should also display the target value counts in absolute values to highlight the small number of occurrences of the minority class: in general it's not possible to do meaningful ML pipeline evaluation with lower than a few hundred datapoints in the minority class.